x-pack/plugins/apm/server/lib/traces/get_trace.ts

```diff
@@ -37,10 +57,46 @@ export async function getTrace(
             }
           ]
         }
+      },
+      aggs: {
+        transactions: {
+          terms: {
+            field: 'transaction.id'
+          },
+          aggs: {
+            errors: {
+              filter: {
+                term: {
+                  'processor.event': 'error'
+                }
+              }
+            }
+          }
+        }
```

Comment:
I just noticed a bug in the existing implementation: we specify `index: config.get('apm_oss.transactionIndices')` meaning that we only query the indices containing transactions. By default this points to `apm*` so is not a problem, but if the user starts using the feature like `apm-*-transactions-*` it will not work.

We could solve this by simply querying all indices:
```ts
    index: [
      config.get('apm_oss.errorIndices'),
      config.get('apm_oss.spanIndices'),
      config.get('apm_oss.transactionIndices')
    ]
```

However, I don't think that is the best approach. The waterfall should only contain transactions and spans, and then we can do an aggregation to get the errors.
Right now the waterfall contain the actual error documents, which can be quite big with stack traces and all - which we don't even need.

I therefore propose limiting the current query to transactions and spans:
```
    index: [
      config.get('apm_oss.spanIndices'),
      config.get('apm_oss.transactionIndices')
    ],
```

And then doing a separate query to get the error counts from the error indices.